nixos/modules/services/networking/peroxide.nix

```diff
@@ -0,0 +1,134 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.peroxide;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  options.services.peroxide = {
+    enable = mkEnableOption "peroxide service";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.peroxide;
+      defaultText = literalExpression "pkgs.peroxide";
+      description = "Which peroxide package to use.";
+    };
+
+    selfSignedCert = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to generate a self-signed X.509 certificate on first run.
+
+          If this option is disabled, then you must provide your own certificate
+          and place it in the service's configuration directory for the peroxide service
+          to start.
+        '';
+      };
+
+      organization = mkOption {
```

Comment:
If the user changes this option and rebuilds their system what happens? Something entirely not reproducible.

This option has questionable value. I suggest we just set a hard coded value and if the user needs something different they are free to provide their own certificate.

Reply:
Done.